amount_helper.rb

```diff
@@ -0,0 +1,54 @@
+module AmountHelper
+  def amount(number, options = {})
+    if number.nil? && options[:nil_as_dash] != false
+      ndash
+    elsif number.zero? && options[:zero_as_dash]
+      ndash
+    elsif number.present?
+      precision = options[:precision]
+      if precision
+        options[:minimum_precision] = precision
+        options[:maximum_precision] = precision
+      end
+
+      number = number.round if options[:round]
+      separator = options.fetch(:separator, I18n.t('number.format.separator'))
+      delimiter = options.fetch(:delimiter, I18n.t('number.format.delimiter')).to_s
+      minimum_precision = options[:minimum_precision] || 0
+      str = number.to_s.sub(/\.0+\z/, '')
+      str =~ /\A(\-?)(\d+)(?:\.(\d+))?\z/ || raise("Could not parse number: #{number}")
+      sign = Regexp.last_match(1)
+      integer = Regexp.last_match(2)
+      fraction = (Regexp.last_match(3) || '')
+
+      if options[:maximum_precision]
+        fraction = fraction[0, options[:maximum_precision]] if options[:maximum_precision]
+      end
+
+      if minimum_precision > 0
+        if !fraction.empty? || minimum_precision > 0
+          fraction = "#{fraction}#{'0' * [0, minimum_precision - fraction.length].max}"
+        end
+      end
+
+      # the following two lines appear to be the most performant way to add a delimiter to every thousands place in the number
```

Comment:
This will achieve the same result but faster:
 str = integer.reverse.gsub(/(\d{3})(?=\d)/, "\\1#{delimiter}").reverse